module3/labs/week1/HowTheWebWorks.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: page
+title: How the Web Works Lab
+---
+
+<aside class="instructor-notes">
+    <p><strong>Instructor Note</strong><br>This lab should be an individual or small group exploration.  Before breaking for worktime, show students how to open their Comman Prompt so that they can try out `curl`</p>
+</aside>
+
+## Lab Objectives
+* Reinforce understanding of the HTTP request/response cycle
+* Use a basic client to see the cycle in action
+
+## Requests and Responses in Detail
+## HTTP Requests and Responses
+
+The HyperText Transfer Protocol gives us rules about how messages should be sent around the Internet. The system that initiates a connection sends a "request", and the system the answers sends a "response".
+
+### HTTP Request
+
+When a "client" (like a web browser) retrieves information, it sends a payload of data to a server as a "request". This request is made up of three main parts:
+
+- A Request line, containing three piece of information:
+  - the HTTP **verb** (also called an http method) for sending or retrieving information
+  - the URI **path** of the resource where we're sending or retrieving information
+  - the version of the HTTP protocol our "client" software is using, usually `HTTP/1.1`
+
+- Headers, which are key/value pairs, which contain supplemental information about our request
+
+- An optional body; we only send data to the server in the body when we are creating or modifying something
+
+### HTTP Response
+
+When the server or web application is finished processing our request, it will send back a response which is a payload of data, and is made up of three main parts:
+
+- a Status line, containing three pieces of information:
+  - The version of the HTTP protocol that this response is using
+  - a 3-digit numeric "status code"
+  - a user-friendly string description of what the "status code" means
+
+- Headers, also sent as key/value pairs similar to the HTTP request
+
+- An optional body; almost all responses will contain additional data in the body. In mod 2, our "body" payload will almost always be HTML.
+
+## Seeing Request/Response in action
+Let's open [Command Prompt](https://www.makeuseof.com/tag/a-beginners-guide-to-the-windows-command-line/) and run some commands to connect to Google's home page.
+
+Enter `curl google.com -v` in a terminal window and review the output.
```

Comment:
What do you think about doing the demonstration in the browser and using the chrome dev tools to look at the request and response? Might not be worth changing, but I'm not sure how much introducing curl helps here.

Reply:
<img width="696" alt="Screenshot 2023-04-12 112700" src="https://user-images.githubusercontent.com/11747682/231536643-e50a37e0-8071-4ab5-9786-254b21c816b9.png">
Maybe showing a request like this is the example

Reply:
Also I definitely make sure to have students only show `Fetch/XHR` requests so it's not quite as overwhelming to look at the network tab.

Reply:
<img width="672" alt="Screenshot 2023-04-12 113748" src="https://user-images.githubusercontent.com/11747682/231539074-c244bb26-7db8-4fea-88d6-c91109150d6f.png">

Liking a youtube video is my favorite POST example to show students in the browser.

Reply:
Since we're not a frontend course it feels a bit less relevant to dive into dev tools, so we can take it or leave it. But I am leaning towards it being helpful to really visualize what's happening and make good connections to websites they visit in their daily life.

Reply:
I agree that the dev tools is helpful - but I would like to introduce them more as a FE tool like in the HTML/CSS lessons.  

What I really like about `curl` is that it helps students see that a 'client' does not always have to be a browser.  Unless you have **strong** feelings about it, I would prefer to try it this way and adjust from there if it totally bombs.